### ISSUE TITLE
No signature for functools partialmethod

### ISSUE THREAD

**Issue by ffe4** [davidhalter/jedi#1519]
`Script.get_signatures()` does not return function signatures for `functools.partialmethod`. Same issue was raised and fixed for `functools.partial` #1371

example:
```python
import jedi

partialmethod = '''
import functools

class X:
    def a(self, n, m):
        print(n, m)
    
    b = functools.partialmethod(a, 1)

X().a(1, 2)
X().b(2)
'''

script = jedi.Script(partialmethod)
print(script.get_signatures(line=10, column=len('X().a(')))
print(script.get_signatures(line=11, column=len('X().b(')))
```
output:
```
[<Signature: index=0 a(n, m)>]
[]
```

**Comment by davidhalter** [comment]
Thanks for the reproduction! That helps a lot. I'm pretty sure this is a bug (probably a small one surrounding methods).

**Comment by davidhalter** [comment]
Oh now I realized. This is not really a bug, it's just a missing feature :) 

partialmethod is not implemented in Jedi, so Jedi does not know what to do with it :)

If you want to work on this (you're very welcome!) search for `functools_partial` in `jedi/plugins/stdlib.py` and implement `functools_partialmethod` in a similar way.

**Comment by ffe4** [comment]
Seems manageable, I'll give it a try over the weekend.

**Comment by davidhalter** [comment]
Tests are in `test/completion/stdlib.py` and you can run it with `pytest test/test_integration.py -k stdlib`.

**Comment by ffe4** [comment]
At first glance I thought I’d only be getting a signature, but this seems a little more involved after all.

While I’ve managed to modify the `partial` implementation to give me a signature for `partialmethod` with marginal effort, but that seems to be a [hack at best](https://github.com/ffe4/jedi/commit/873d85f064d27d8c9b0c6082bc5ef64282751d9d#commitcomment-37698690). I’m still not quite sure how methods and functions are handled differently and how the `py__call__` method works. 

I’ll have to look into it a bit more, but if you have some pointers regarding where to look they’d be appreciated. I’m probably not yet at a level to work on a project like this though, to be completely honest, but I’m happy to keep trying.

I’d also be happy to just have it working, however, so I don’t want to stop someone who knows what they are doing from implementing it either.

**Comment by davidhalter** [comment]
That doesn't look bad I guess. Obviously you shouldn't disable the partial implementation, but other than that it's fine I guess.

You just have to add a few tests and if there's issues, I will try to support you.

**Comment by ffe4** [comment]
I think I was not clear enough in my comment on the commit. 

While I did not have time yet to look further into it, I have submitted a draft pull request with the implementation and tests as I have them right now. I have also written tests in `test/test_inference/test_signature.py`.

I have also attached the failed test report, so I hope it's clearer this time. I think that I have to do a more method-like implementation to fix both issues mentioned in the PR, although I'm not sure yet what that means.
